src/Controller/UserController.php

```diff
@@ -70,45 +71,43 @@ public function getUserDetails(User $user): Response
     }
 
     /**
-     * Create an User.
+     * Modify an User.
      * @Rest\Post("/{id}/edit")
-     * @param SerializerInterface $serializer
+     * @param User $user
      * @param Request $request
      * @param UserRepository $userRepository
      * @return JsonResponse|Response
      * @throws ORMException
      * @throws OptimisticLockException
      */
-    public function editUser($id, SerializerInterface $serializer, Request $request, UserRepository $userRepository)
+    public function editUser(User $user, Request $request, UserRepository $userRepository)
     {
-        $user = $userRepository->find($id);
-        if (!$user) {
-            return new JsonResponse(['message' => 'The user was not found!'], Response::HTTP_NOT_FOUND);
+        $authenticatedUser = $this->getUser();
+
+        if ($authenticatedUser->getId() !== $user->getId()) {
+            return new JsonResponse(['message' => 'Access denied!'], Response::HTTP_BAD_REQUEST);
```

Comment:
Actualy it should be `Response::HTTP_FORBIDDEN` as the current user cannot edit this user. You can fix it after the merge.